jetty-websocket/websocket-core/src/main/java/org/eclipse/jetty/websocket/core/client/HttpClientProvider.java

```diff
@@ -39,15 +39,17 @@ static HttpClient get()
             Log.getLogger(HttpClientProvider.class).ignore(ignore);
         }
 
-        return new HttpClientProvider(){}.newHttpClient();
+        return HttpClientProvider.newDefaultHttpClient();
     }
 
-    default HttpClient newHttpClient()
+    private static HttpClient newDefaultHttpClient()
     {
         HttpClient client = new HttpClient(new SslContextFactory());
         QueuedThreadPool threadPool = new QueuedThreadPool();
         threadPool.setName("WebSocketClient@" + client.hashCode());
         client.setExecutor(threadPool);
         return client;
     }
+
+    HttpClient newHttpClient();
```

Comment:
This method should be implemented by calling `newDefaultHttpClient()`.
In this way, subclasses can call `super`, obtain the default instance and further customize it.